dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -2611,225 +2612,65 @@ public boolean isContentEqual(Contentlet contentlet1,Contentlet contentlet2, Use
-                contentRelationships = findContentRelationshipsFromIndex(workingCon);
-            } else {
-                contentRelationships = findContentRelationshipsFromIndex(contentlet);
-            }
-        }
-        else
-        {
-            contentRelationships = findContentRelationshipsFromIndex(contentlet);
-        }
-
-        if(permissions == null)
-            permissions = new ArrayList<Permission>();
-        if(cats == null)
-            cats = new ArrayList<Category>();
-        if(contentRelationships == null)
-            contentRelationships = new HashMap<Relationship, List<Contentlet>>();
-        if(workingCon == null)
-            workingCon = contentlet;
-        return checkin(contentlet, contentRelationships, cats, permissions, user, respectFrontendRoles, false);
+    public Contentlet checkin(Contentlet contentlet, User user,boolean respectFrontendRoles)
+            throws IllegalArgumentException,DotDataException, DotSecurityException {
+        return checkin(contentlet, (ContentletRelationships) null, null, null, user,
+            respectFrontendRoles, false);
     }
 
     @CloseDBIfOpened
     @Override
-    public Contentlet checkin(Contentlet contentlet, User user,boolean respectFrontendRoles, List<Category> cats)throws IllegalArgumentException, DotDataException,DotSecurityException, DotContentletStateException,DotContentletValidationException {
-
-        List<Permission> permissions = null;
-        Map<Relationship, List<Contentlet>> contentRelationships = null;
-        Contentlet workingCon = null;
-
-        //If contentlet is not new
-        if(InodeUtils.isSet(contentlet.getIdentifier())) {
-            workingCon = findWorkingContentlet(contentlet);
-            if(workingCon != null){
-                if(cats==null) {
-                    cats = categoryAPI.getParents(workingCon, APILocator.getUserAPI().getSystemUser(), true);
-                }
-                permissions = permissionAPI.getPermissions(workingCon, false, true);
-                contentRelationships = findContentRelationships(workingCon);
-            } else {
-                contentRelationships = findContentRelationships(contentlet);
-            }
-        } else {
-            contentRelationships = findContentRelationships(contentlet);
-        }
-
-        if(permissions == null)
-            permissions = new ArrayList<Permission>();
-        if(cats == null)
-            cats = new ArrayList<Category>();
-        if(contentRelationships == null)
-            contentRelationships = new HashMap<Relationship, List<Contentlet>>();
-        if(workingCon == null)
-            workingCon = contentlet;
-        return checkin(contentlet, contentRelationships, cats, permissions, user, respectFrontendRoles);
+    public Contentlet checkin(Contentlet contentlet, User user,boolean respectFrontendRoles, List<Category> cats)
+        throws IllegalArgumentException, DotDataException,DotSecurityException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)